### PR TITLE
More achievement UI fixes

### PR DESCRIPTION
**Pull request by hrydgard**
Pretty self-explanatory stuff, see the commits.

Part of #17631

### CODE REVIEW

File: Common/Render/Text/draw_text_win.cpp [hunk 1]
```diff
@@ -134,13 +134,25 @@ void TextDrawerWin32::MeasureString(const char *str, size_t len, float *w, float
...
+
+		int extW = 0, extH = 0;
+		for (auto &line : lines) {
+			SIZE size;
+			std::wstring wstr = ConvertUTF8ToWString(line);
+			GetTextExtentPoint32(ctx_->hDC, wstr.c_str(), (int)wstr.size(), &size);
+
+			if (size.cx > extW)
+				extW = size.cx;
+			extH += size.cy;
```
Comment (unknownbrackets) [inline Common/Render/Text/draw_text_win.cpp:150]: Note that MeasureStringRect already doing something similar.  It's thing with a space for blank lines might be important, and it handles leading more correctly.  I think at one point, at least, the intent was to use it for multiline strings...

-[Unknown]

Comment (hrydgard) [reply Common/Render/Text/draw_text_win.cpp:150]: MeasureStringRect does the wrappy thing, and caches each line measurement individually. Although hm, yeah, it's only an option in MeasureStringRect, so maybe should have used that with dummy bounds.

All the other text backends actually compute the height correctly even with MeasureString, though.

Comment (unknownbrackets) [reply Common/Render/Text/draw_text_win.cpp:150]: Well, at least should use the correct leading calculation and the space for blank lines.  Putting aside text wrap.

-[Unknown]

Comment (hrydgard) [reply Common/Render/Text/draw_text_win.cpp:150]: Hm, fair, I missed that part. Little trigger happy with the merge, merged before I saw this comment.

I might just go back on this and instead use MeasureStringRect.

**Pull request merged**